### DISCUSSION TITLE
is there any way to move the options menu to a separate menu with out breaking the game

### DISCUSSION THREAD

**Discussion by Raptorq1** [Techokami/SonicWorldsNext#137] [Q&A] [answered]
is there any way to move the options menu to a separate menu with out breaking the game I am asking this question because my fan game is inspired by sonic mania and sonic mania doesn't have options on the pause menu

**Comment by r6915ee** [comment]
Maybe you could add an instance of the pause menu to your main menu that gets activated on a certain menu button push or what you're using?

**Reply by Raptorq1** [reply]
I got it to show but there's no input

**Reply by r6915ee** [reply]
You could maybe try checking any mentions of the pause menu being used and see how those instances do it.

**Reply by Raptorq1** [reply]
I am using separate scene for the options menu

**Reply by Raptorq1** [reply]
It doesn't instance it

**Reply by r6915ee** [reply]
If it's a separate scene then you could probably use change_scene_to_file() if it works similar to the Mania options menu.

**Reply by Raptorq1** [reply]
I have done that 

Here's the code For the scene switching 


func _input(event):
	if !selected:
		# select character rotation
		if Input.is_action_just_pressed("gm_left"):
			MenuID = wrapi(MenuID-1,0,MenuLabels.size())
		if Input.is_action_just_pressed("gm_right"):
			MenuID = wrapi(MenuID+1,0,MenuLabels.size())
		$CanvasLayer2/Control/Level.text = MenuLabels[MenuID]
		if Input.is_action_just_pressed("gm_action"):
			match(MenuID):
				0: # Sonic and Tails
					$CanvasLayer3.show()
					$Node2D/AudioStreamPlayer2.play()
					$CanvasLayer3/Top/Transition.play("In")
					await $CanvasLayer3/Top/Transition.animation_finished
					$CanvasLayer3/Top/Transition.play("Out")
					Global.main.change_scene_to_file(sound)
				1: # Sonic
					$CanvasLayer3.show()
					$Node2D/AudioStreamPlayer2.play()
					$CanvasLayer3/Top/Transition.play("In")
					await $CanvasLayer3/Top/Transition.animation_finished
					$CanvasLayer3/Top/Transition.play("Out")
					Global.main.change_scene_to_file(Extras)
				2: # Sonic
					$CanvasLayer3.show()
					$Node2D/AudioStreamPlayer2.play()
					$CanvasLayer3/Top/Transition.play("In")
					await $CanvasLayer3/Top/Transition.animation_finished
					$CanvasLayer3/Top/Transition.play("Out")
					Global.main.change_scene_to_file(Extras)
				3: # Sonic
					$CanvasLayer3.show()
					$Node2D/AudioStreamPlayer2.play()
					$CanvasLayer3/Top/Transition.play("In")
					await $CanvasLayer3/Top/Transition.animation_finished
					$CanvasLayer3/Top/Transition.play("Out")
					Global.main.change_scene_to_file(Extras)
		if Input.is_action_just_pressed("gm_action2"):
			$CanvasLayer3/Top/Transition/Woosh.play()
			Global.main.change_scene_to_file(back)

**Reply by Raptorq1** [reply]
Even by doing what I just showed you there's still no inputs

**Reply by r6915ee** [reply]
Then it could be the way its input is handled. I haven't delved into this yet, but if you can only control just the main menu and not the options menu, then maybe the main menu is just overriding it? I haven't checked this out yet so if you can perhaps send a video as well then please do.

**Reply by Raptorq1** [reply]
All the menu options are separate scenes

**Reply by r6915ee** [reply]
Then was something like `event.is_action_released("gm_left")` featured in the options menu script?

**Reply by Raptorq1** [reply]
There isn't any event.is_action_released("gm_left") in the options menu script

**Reply by r6915ee** [reply]
Then I'd suggest maybe adding some of them in the right places; I'd recommend looking at the project's input settings and maybe tinkering with their actions as an if statement.

**Reply by Raptorq1** [reply]
I fixed it

**Reply by Raptorq1** [reply]
If I remove The options Manu from the pause menu restart option and quit don't work

**Reply by r6915ee** [reply]
Did you remove the "options" button from menusText? If so, then you should probably remove the options button from the MENUS enum as well. And maybe change up some of the menu ids to consider the removed options button, just in case.

**Reply by Raptorq1** [reply, marked as answer]
I have fixed it
